ui/plugin-system/src/components/PluginKindSelect/PluginKindSelect.tsx

```diff
@@ -12,31 +12,67 @@
 // limitations under the License.
 
 import { MenuItem, TextField, TextFieldProps } from '@mui/material';
-import { forwardRef } from 'react';
+import { forwardRef, useCallback } from 'react';
 import { PluginType } from '../../model';
 import { useListPluginMetadata } from '../../runtime';
+import { PluginEditorSelection } from '../PluginEditor';
 
-export interface PluginKindSelectProps extends Omit<TextFieldProps, 'children'> {
-  pluginType: PluginType;
+export interface PluginKindSelectProps extends Omit<TextFieldProps, 'value' | 'onChange' | 'children'> {
+  pluginTypes: PluginType[];
+  value?: PluginEditorSelection;
+  onChange?: (s: PluginEditorSelection) => void;
 }
 
 /**
- * Displays a MUI Select input for selecting a plugin's kind from a list of all the available plugins of a specific
- * plugin type. (e.g. "Show a list of all the Panel plugins" or "Show a list of all the Variable plugins").
+ * Displays a MUI Select input for selecting a plugin's kind from a list of all the available plugins of some specific
+ * plugin types. (e.g. "Show a list of all the Panel plugins", or "Show a list of all the Variable plugins", or "Show
+ * a list of all the TimeSeriesQuery, TraceQuery, and LogQuery plugins").
+ * The value of the select is the kind of the plugin, but you can also listen to the `onPluginTypeChange` event to know
+ * when the user changes the plugin type (it fires at start for the default value.)
  */
 export const PluginKindSelect = forwardRef((props: PluginKindSelectProps, ref) => {
-  const { pluginType, value: propValue, ...others } = props;
-  const { data, isLoading } = useListPluginMetadata(pluginType);
+  const { pluginTypes, value: propValue, onChange: propOnChange, ...others } = props;
+  const { data, isLoading } = useListPluginMetadata(pluginTypes);
 
   // Pass an empty value while options are still loading so MUI doesn't complain about us using an "out of range" value
-  const value = propValue !== '' && isLoading ? '' : propValue;
+  const value = propValue && isLoading ? '' : JSON.stringify(propValue);
```

Comment:
I’m not totally following these JSON.stringify lines 🤔

Reply:
Well, you took the last PR of the thread, but there are three commits, each depending on the previous, and each commit having its own PR ^^'
This is being discussed here: https://github.com/perses/perses/pull/1887#discussion_r1557670040

Reply:
😅 lol sorry! Guess I should have read that matrix message before jumping in to review, I’ll try to find time to start at the beginning of the stack but definitely don’t wait on me, looks like Gabriel is giving a more thorough review anyways

Reply:
I’ve started using Graphite lately btw, makes it easy to follow along with stacked PRs, although probably wouldn’t have helped me here since I should have just read your PR description 🤦‍♂️